IPython/html/widgets/widget_output.py

```diff
@@ -38,13 +38,30 @@ def clear_output(self, *pargs, **kwargs):
             clear_output(*pargs, **kwargs)
 
     def __enter__(self):
+        """Called upon entering output widget context manager."""
         self._flush()
-        self.send({'method': 'push'})
+        kernel = get_ipython().kernel
+        session = kernel.session
+        send = session.send
+        self._original_send = send
+        self._session = session
+
+        def send_hook(stream, msg_or_type, *args, **kwargs):            
+            if stream is kernel.iopub_socket and msg_or_type in ['clear_output', 'stream', 'display_data']:
```

Comment:
I'd be inclined to handle the possibility anyway, because the method you're replacing allows that possibility, and even if none of the code currently uses it, it may in the future. And it should be an easy couple of lines to handle it.